server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/Rights.java

```diff
@@ -0,0 +1,243 @@
+        }
+
+        public static Right forChar(char c) {
+            return Arrays.stream(values())
+                .filter(right -> right.asCharacter() == c)
+                .findAny()
+                .orElseThrow(() -> new IllegalArgumentException("No matching right for '" + c + "'"));
+        }
+    }
+
+    public static class Username {
+        private final String value;
+
+     //   @JsonCreator
+        public Username(String value) {
+            this.value = value;
+        }
+
+        @JsonValue
+        public String getValue() {
+            return value;
+        }
+
+        @Override
+        public final boolean equals(Object o) {
+            if (o instanceof Username) {
+                Username username = (Username) o;
+
+                return Objects.equals(this.value, username.value);
+            }
+            return false;
+        }
+
+        @Override
+        public final int hashCode() {
+            return Objects.hash(value);
+        }
+    }
+
+    public static class Builder {
+        private Multimap<Username, Right> rights;
+
+        public Builder() {
+            rights = ArrayListMultimap.create();
+        }
+
+        public Builder delegateTo(Username username, Right... rights) {
+            delegateTo(username, Arrays.asList(rights));
+            return this;
+        }
+
+        public Builder delegateTo(Username username, Collection<Right> rights) {
+            this.rights.putAll(username, rights);
+            return this;
+        }
+
+        public Builder combine(Builder builder) {
+            this.rights.putAll(builder.rights);
+            return this;
+        }
+
+        public Rights build() {
+            return new Rights(rights);
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static Rights fromACL(MailboxACL acl) {
+        return acl.getEntries()
+            .entrySet()
+            .stream()
+            .filter(entry -> isSupported(entry.getKey()))
+            .map(Rights::toRightsBuilder)
+            .reduce(builder(), Builder::combine)
+            .build();
+    }
+
+    private static Builder toRightsBuilder(Map.Entry<EntryKey, MailboxACL.Rfc4314Rights> entry) {
+        return builder().delegateTo(
+            new Username(entry.getKey().getName()),
+            fromACL(entry.getValue()));
+    }
+
+    private static List<Right> fromACL(MailboxACL.Rfc4314Rights rights) {
+        return rights.list()
+            .stream()
+            .flatMap(right -> OptionalUtils.toStream(Right.forRight(right)))
+            .collect(Guavate.toImmutableList());
+    }
+
+    private static boolean isSupported(EntryKey key) {
+        if (key.isNegative()) {
+            LOGGER.warn("Negative keys are not supported");
+            return false;
+        }
+        if (key.getNameType() != MailboxACL.NameType.user) {
+            LOGGER.warn(key.getNameType() + " is not supported. Only 'user' is.");
```

Comment:
i'm sure I don't want to have a lot of warning in production because of that

Reply:
Let's put it in info